### PR TITLE
fix: [CDS-86541]: if gitX enforced true then select remote store type by default

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- if enforce gitX setting is true then by default we would select storeType === Remote and disable the inline option , currently we had the dependency missing on settings value as it comes via API so updated that

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/9acca5fd-14c7-4eba-8512-b6c1f002add4

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by vikrant-harness** [comment]
trigger messagecheck

File: src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx [hunk 1]
```diff
@@ -158,7 +158,7 @@ export default function CreatePipelines({
       })
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isEdit])
+  }, [isEdit, isGitXEnforced])
```
Comment (sunnykesh-harness) [inline src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx:161]: **isGitXEnforced** is not used in this useEffect, should not we use **initialValues.storeType** here?

Comment (sunnykesh-harness) [reply src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx:161]: Also check line 104, we are handling isGitXEnforced there.

Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx:161]: the parent passes isGitXEnforced to the component and initially till the API call is not success it comes as false and the queryParams are updated with inline , so when it updates we update the queryParams, and as intialValues.storeType also depends on same in parent so ideally there is no difference.

**Review by Sagar-Rajput7 — DISMISSED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by litvinovstas — APPROVED** [review]

**Pull request merged**